modules/fbs/includes/fbs.loan.inc

```diff
@@ -48,13 +48,22 @@ function fbs_loan_list($account, $reset = FALSE) {
         $id = $loan->loanDetails->loanId;
 
         $loan_data = array(
-          'ding_entity_id' => $ids[$loan->loanDetails->recordId],
           'loan_date' => $loan->loanDetails->loanDate,
           'expiry' => $loan->loanDetails->dueDate,
           'renewable' => $loan->isRenewable,
           'materials_number' => $loan->loanDetails->materialItemNumber,
         );
 
+        // With the FBS provider we have to use FBS data for ILL, since we can't
+        // rely on the record ID being unique in these cases.
+        if (!empty($loan->loanDetails->ilBibliographicRecord)) {
+          $loan_data['display_name'] = $loan->loanDetails->ilBibliographicRecord->title;
+          $loan_data['author'] = $loan->loanDetails->ilBibliographicRecord->author;
+        }
+        elseif (isset($ids[$loan->loanDetails->recordId])) {
```

Comment:
@holt83 Lots of code seems to rely in the presence of `ding_entity_id`. In this clause you might be in a situation where this critical value is not set., for example here: https://github.com/ding2/ding2/blob/master/modules/p2/ding_serendipity/ding_serendipity.module#L419-L437.

Maybe this line should not be wrapped in an `elseif`?

Reply:
I think that ding_serendipity should check for an entity like for example ding_loan does and just skip the loan if it's not present. It means the provider didn't return an entity for some reason. With FBS as provider it's because the loan is an ILL and it should not be on any ding_list and therefore ding_serendipity should ignore it. 

Example here: https://github.com/ding2/ding2/blob/master/modules/ding_loan/plugins/content_types/loans.inc#L179